storage/entities/peer.js

```diff
@@ -0,0 +1,304 @@
+		);
+
+		const params = {
+			limit: parsedOptions.limit,
+			offset: parsedOptions.offset,
+			parsedFilters,
+		};
+
+		return this.adapter.executeFile(
+			this.SQLs.select,
+			params,
+			{ expectedResultCount },
+			tx
+		);
+	}
+
+	/**
+	 * Create peer object
+	 *
+	 * @param {Object} data
+	 * @param {Object} [_options]
+	 * @param {Object} tx - Transaction object
+	 * @return {null}
+	 */
+	// eslint-disable-next-line no-unused-vars
+	create(data, _options = {}, tx = null) {
+		const objectData = defaults(data, defaultCreateValues);
+		const createSet = this.getValuesSet(objectData);
+		const attributes = Object.keys(data)
+			.map(k => `"${this.fields[k].fieldName}"`)
+			.join(',');
+
+		return this.adapter.executeFile(
+			this.SQLs.create,
+			{ createSet, attributes },
+			{ expectedResultCount: 0 },
+			tx
+		);
+	}
+
+	/**
+	 * Update the records based on given condition
+	 *
+	 * @param {filters.Peer} [filters]
+	 * @param {Object} data
+	 * @param {Object} [options]
+	 * @param {Object} tx - Transaction object
+	 * @return {null}
+	 */
+	update(filters, data, options = {}, tx = null) {
+		this.validateFilters(filters);
+		this.validateOptions(options);
+		const objectData = omit(data, readOnlyFields);
+		const mergedFilters = this.mergeFilters(filters);
+		const parsedFilters = this.parseFilters(mergedFilters);
+		const updateSet = this.getUpdateSet(objectData);
+
+		const params = {
+			...objectData,
+			...{
+				parsedFilters,
+				updateSet,
+			},
+		};
+
+		return this.adapter.executeFile(
+			this.SQLs.update,
+			params,
+			{ expectedResultCount: 0 },
+			tx
+		);
+	}
+
+	/**
+	 * Update one record based on the condition given
+	 *
+	 * @param {filters.Peer} filters
+	 * @param {Object} data
+	 * @param {Object} [options]
+	 * @param {Object} tx - Transaction object
+	 * @return {null}
+	 */
+	updateOne(filters, data, options = {}, tx = null) {
+		this.validateFilters(filters);
+		this.validateOptions(options);
+		const objectData = omit(data, readOnlyFields);
+		const mergedFilters = this.mergeFilters(filters);
+		const parsedFilters = this.parseFilters(mergedFilters);
+		const updateSet = this.getUpdateSet(objectData);
+
+		const params = {
+			...objectData,
+			...{
+				parsedFilters,
+				updateSet,
+			},
+		};
+
+		return this.adapter.executeFile(
+			this.SQLs.updateOne,
+			params,
+			{ expectedResultCount: 0 },
+			tx
+		);
+	}
+
+	/**
+	 * Check if the record exists with following conditions
+	 *
+	 * @param {filters.Peer} filters
+	 * @param {Object} [options]
+	 * @param {Object} [tx]
+	 * @returns {Promise.<boolean, Error>}
+	 */
+	isPersisted(filters, options = {}, tx = null) {
+		const atLeastOneRequired = true;
+		this.validateFilters(filters, atLeastOneRequired);
+		this.validateOptions(options);
+		const mergedFilters = this.mergeFilters(filters);
+		const parsedFilters = this.parseFilters(mergedFilters);
+
+		return this.adapter
+			.executeFile(this.SQLs.isPersisted, { parsedFilters }, {}, tx)
```

Comment:
Pass `{ expectedResultCount: 1 }` to make sure it don't return array.

Reply:
Done. Resolving.